mitosheet/src/components/taskpanes/UpdateImports/UpdateImportsUtils.tsx

```diff
@@ -0,0 +1,74 @@
+import { UIState } from "../../../types";
+import { DataframeImportParams } from "../DataframeImport/DataframeImportTaskpane"
+import { CSVImportParams } from "../Import/CSVImport";
+import { ExcelImportParams } from "../Import/XLSXImport";
+import { TaskpaneType } from "../taskpanes";
+import { UpdatedImportObj } from "./UpdateImportsTaskpane"
+
+export interface UpdateImportData {
+    updatedImportObjs: UpdatedImportObj[],
+    importIndex: number
+}
+
+
+export const updateImportedDataWithDataframe = (
+    updateImportedData: UpdateImportData | undefined, 
+    newImportParams: DataframeImportParams,
+    setUIState: React.Dispatch<React.SetStateAction<UIState>>
+): void => {
+    if (updateImportedData === undefined) {
+        return 
+    }
+
+    const newUpdatedImportObjs: UpdatedImportObj[] = JSON.parse(JSON.stringify(updateImportedData.updatedImportObjs))
+    const importIndex = updateImportedData.importIndex
+
+    newUpdatedImportObjs[importIndex] = {
+        ...newUpdatedImportObjs[importIndex],
+        type: 'df', 
+        import_params: newImportParams
+    }
+
+    setUIState(prevUIState => {
+        return {
+            ...prevUIState,
+            currOpenTaskpane: {type: TaskpaneType.UPDATEIMPORTS, updatedImportObjs: newUpdatedImportObjs}
```

Comment:
Oh. I see what's happening here. Different taskpanes get opened, which then requires you to reopen this taskpane, so you're passing through the global state. That's messy.

I think the DataframeImportScreen approach would generally avoid all of this, and allow you to just get the params back directly from the screen. It would make the UpdateImportsTaskpane a simple tree, rather than a taskpane that is repeatedly opened and closed -- which is much better and long-term maintainable.